### PR TITLE
Added collection duplication and junit reporter

### PR DESCRIPTION
**Pull request by SanskritiHarmukh**
This PR includes feature additions related to Release v2024.7.0:

- [x] Collection Duplication
- [x] JUnit Reporter

### CODE REVIEW

File: documentation/clients/cli/overview.mdx [hunk 1]
```diff
@@ -89,6 +89,30 @@ hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [-
...
+| `<failure>`                   | Indicates an assertion failure within a `<testcase>`. Includes type and message attributes describing the failure.                                                                      |
+| `<error>`                     | Indicates an error during assertion within a `<testcase>`. Includes message attribute describing the error.                                                                             |
+| `time`                        | Attribute of `<testsuite>` that shows the execution time for the test cases (excluding request execution time). The total time is at the root `<testsuites>`.                           |
+| `timestamp`                   | Attribute of `<testsuite>` that records the execution date and time in ISO string format.                                                                                               |
+| `tests`, `failures`, `errors` | Attributes of `<testsuite>` that track the number of test cases, failed cases, and errors, respectively. Set to 0 if errors halt further execution.                                     |
```
Comment (jamesgeorge007) [inline documentation/clients/cli/overview.mdx:115]:
```suggestion
| `tests`, `failures`, `errors` | Attributes of `<testsuite>` and `testsuites` that track the number of test cases, failed cases, and errors, at the request level and effective count at the root level test suite respectively. Set to `0` at a request level test suite, if errors halt further execution.                                     |
```

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
We can also add an entry for the `system-err` element under `JUnit Report Format Overview` that indicates errors reported at the request level.

> For errors reported at the request level - invalid URL, reference error in the test script, etc are represented by the `system-err` element with more information available under `CDATA`. It is to be noted that the report follows a convention where errors reported at the request level are compiled in a single `CDATA` element separated by newlines.

File: documentation/clients/cli/overview.mdx [hunk 2]
```diff
@@ -89,6 +89,31 @@ hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [-
...
+| `<testsuite>`                 | Represents a collection of test cases for a specific request.                                                                                                                                                                                                                 |                                       
+| `<testcase>`                  | Represents an individual test case. It corresponds to `pw.expect()` assertions with the description prefixed by that of the test suite `pw.test()`.  It appears as a direct child of `<testsuite>`.                                                                           |
+| `name`	                      | Indicates the hierarchy of collections up to the request. Organized at the request level using the naming convention: `<parent-collection-name>/<child-collection-name>/<request-name>`.                                                                                      |
```
Comment (jamesgeorge007) [inline documentation/clients/cli/overview.mdx:109]:
```suggestion
| `name`	                      | Attribute for For `testsuite`, it indicates the hierarchy of collections up to the request. Organized at the request level using the naming convention: `<parent-collection-name>/<child-collection-name>/<request-name>`. While, for `testcase`, it'll be the description.                                             
```

File: documentation/clients/cli/overview.mdx [hunk 3]
```diff
@@ -89,6 +89,31 @@ hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [-
...
+| `<failure>`                   | Indicates an assertion failure within a `<testcase>`. Includes `type` and `message` attributes describing the failure.                                                                                                                                                        |
+| `<error>`                     | Indicates an error during assertion within a `<testcase>`. Includes `message` attribute describing the error.                                                                                                                                                                 |
+| `system-err`                  | Represents errors reported at the request level (e.g., invalid URL, reference error in the test script).  These errors are detailed within a `CDATA` section, with each error separated by newlines to ensure each issue is clearly identified.                                 |
```
Comment (jamesgeorge007) [inline documentation/clients/cli/overview.mdx:113]:
```suggestion
| `<system-err>`                  | Represents errors reported at the request level (e.g., invalid URL, reference error in the test script).  These errors are detailed within a `CDATA` section, with each error separated by newlines to ensure each issue is clearly identified.                                 |
```

**Review by jamesgeorge007 — APPROVED** [review]

File: documentation/clients/cli/overview.mdx (hunk 3 above)
Comment (SanskritiHarmukh) [reply documentation/clients/cli/overview.mdx]: My bad.

File: documentation/clients/cli/overview.mdx [hunk 4]
```diff
@@ -89,6 +89,31 @@ hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [-
...
+| `<testsuite>`                 | Represents a collection of test cases for a specific request.                                                                                                                                                                                                                 |                                       
+| `<testcase>`                  | Represents an individual test case. It corresponds to `pw.expect()` assertions with the description prefixed by that of the test suite `pw.test()`.  It appears as a direct child of `<testsuite>`.                                                                           |
+| `name`	                      | Attribute for For `testsuite`, it indicates the hierarchy of collections up to the request. Organized at the request level using the naming convention: `<parent-collection-name>/<child-collection-name>/<request-name>`. While, for `testcase`, it'll be the description.
```
Comment (jamesgeorge007) [inline documentation/clients/cli/overview.mdx:109]:
```suggestion
| `name`	                      | Attribute for `testsuite` and `testcase` elements. For `testsuite`, it indicates the hierarchy of collections up to the request. Organized at the request level using the naming convention: `<parent-collection-name>/<child-collection-name>/<request-name>`. While, for `testcase`, it'll be the description.                                             
```

**Pull request merged**